ginga/util/iohelper.py

```diff
@@ -87,10 +87,35 @@ def get_fileinfo(filespec, cache_dir='/tmp', download=False):
         idx = match.group(2)
         if ',' in idx:
             hduname, extver = idx.split(',')
-            hduname = hduname.strip()
-            extver = int(extver)
-            idx = (hduname, extver)
-            name_ext = "[%s,%d]" % idx
+            hduname = hduname.strip().upper()
+
+            # User trying to match extver with wildcard!
+            # NOTE: Only use Astropy FITS; Only works on local file.
+            if extver == '*':
```

Comment:
Can we do something with `io_fits` instead?  Something like:
```python
opener = io_fits.get_fitsloader(logger=logger)
try:
    opener.open_file(filespec)
    if opener.hdu_info is not None:
        for hduinfo in opener.hdu_info:
            ...
finally:
    opener.close()
```

Here is an example of the contents of `opener.hdu_info`:
```python
In [11]: opener.hdu_info
Out[11]: 
[{'index': 0, 'name': 'PRIMARY', 'extver': 1, 'htype': 'PrimaryHDU', 'dtype': ''},
 {'index': 1, 'name': 'SCI', 'extver': 1, 'htype': 'ImageHDU', 'dtype': 'float32'},
 {'index': 2, 'name': 'ERR', 'extver': 1, 'htype': 'ImageHDU', 'dtype': 'float32'},
 {'index': 3, 'name': 'DQ', 'extver': 1, 'htype': 'ImageHDU', 'dtype': 'int16'},
 {'index': 4, 'name': 'SCI', 'extver': 2, 'htype': 'ImageHDU', 'dtype': 'float32'},
 {'index': 5, 'name': 'ERR', 'extver': 2, 'htype': 'ImageHDU', 'dtype': 'float32'},
 {'index': 6, 'name': 'DQ', 'extver': 2, 'htype': 'ImageHDU', 'dtype': 'int16'},
 {'index': 7, 'name': 'WCSCORR', 'extver': 1, 'htype': 'BinTableHDU', 'dtype': '[40A, I, 1A, 24A, 24A, 24A, 24A, D, D, D, D, D, D, D, D, 24A, 24A, D, D, D, D, J, 40A, 128A]'},
 {'index': 8, 'name': 'WCSDVARR', 'extver': 1, 'htype': 'ImageHDU', 'dtype': 'float64'},
 {'index': 9, 'name': 'WCSDVARR', 'extver': 2, 'htype': 'ImageHDU', 'dtype': 'float64'},
 {'index': 10, 'name': 'WCSDVARR', 'extver': 3, 'htype': 'ImageHDU', 'dtype': 'float64'},
 {'index': 11, 'name': 'WCSDVARR', 'extver': 4, 'htype': 'ImageHDU', 'dtype': 'float64'}]
```

Reply:
This keeps the abstraction of the fits opener.